Content.Server/Power/EntitySystems/PowerNetSystem.cs

```diff
@@ -20,7 +23,8 @@ public class PowerNetSystem : EntitySystem
         private readonly HashSet<PowerNet> _powerNetReconnectQueue = new();
         private readonly HashSet<ApcNet> _apcNetReconnectQueue = new();
 
-        private readonly Dictionary<PowerNetworkBatteryComponent, float> _lastSupply = new();
+        private readonly Dictionary<PowerNetworkBatteryComponent, (float supply, float storage)> _lastBatteryState = new();
+        private readonly Dictionary<PowerNetworkBatteryComponent, PowerNetBatteryEvent> _lastBatteryEvent = new();
```

Comment:
PJB disliked that I used the dictionary and these should be stored on the component.

Reply:
1. This is consistent with the existing code
2. Fixing the existing code is out of this PR's scope

If I have to get more involved with this, I'd be more inclined to entirely nix the current approach in favor of something that [could support silly live graphs](https://github.com/space-wizards/space-station-14/pull/5046).

Off the top of my head, that would mean having something that maintains running stats in a ringbuffer. After that, systems could instead inspect the ringbuffer instead of depending on the various `PowerNetSystem` events. I'd guess that would also allow more interesting dynamics than "power's on" "power's off".

Reply:
1. like I said the existing code was bad and the new code should use what pjb said
2. see 1

Anything beyond that sounds like refactoring pow3r territory.